src/elm/interval.coffee

```diff
@@ -287,51 +287,50 @@ module.exports.Expand = class Expand extends Expression
-    # per 1 day" results in intervals with the same start and end: { [@2000-01-01, @2000-01-01],
-    # [@2000-01-02, @2000-01-02] }. However "expand [@2000-01-01, @2000-03-01] per 1 month" results
-    # in { [@2000-01-01, @2000-01-31], [@2000-02-01, @2000-02-28] }. This check allows us
-    # to skip the unnecessary date arithmetic in the first case.
-    point_intervals = current_low.add(per.value, per.unit).predecessor().equals(current_low)
-
-    # count is how many items to put in the result - take duration and then add a one for cases
-    # where e.g. the high point ends a month, like 2018-03-31. Then divide by the per value.
-    # We do this to avoid a date comparison operation for each item in the result list.
-    if per.unit == low.getPrecision()
-      count = Math.floor((low.durationBetween(high,per.unit).high + 1) / per.value)
-    # if precisions are equal, we want the high uncertainty (e.g. [2000, 2002] per year should
-    # give a list of 3) else, take the low uncertainty (e.g. [2012-01-01T13:00:00, 2012-01-02T12:59]
-    # per day should give {} since we dont include any uncertain portions)
-    else
-      count = Math.floor((low.durationBetween(high,per.unit).low + 1) / per.value)
 
-    if point_intervals
-      for i in [1..count + 1] #point intervals result in one extra item (see above example)
-        results.push(new dtivl.Interval(current_low, current_low.copy(), true, true))
-        current_low = current_low.add(per.value, per.unit)
-    else
-      for i in [1..count]
-        current_high = current_low.add(per.value, per.unit).predecessor()
-        results.push(new dtivl.Interval(current_low, current_high, true, true))
-        current_low = current_low.add(per.value, per.unit)
-    # because of the +1 in the count declaration, we might have added an extraneous item at the end.
-    # we use 'not sameOrBefore' instead of 'after' since we do not include items in the results
-    # if they just *might* not be after (due to uncertainty)
-    if results.length > 0 and !results[results.length-1].high.sameOrBefore(high)
-      results.pop()
+    low = @truncateToPrecision(low, per.unit)
+    high = @truncateToPrecision(high, per.unit)
+
+    current_high = current_low.add(per.value, per.unit).predecessor()
+    intervalToAdd = new dtivl.Interval(current_low, current_high, true, true)
+    while intervalToAdd.high.sameOrBefore(high)
+      results.push(intervalToAdd)
+      current_low = current_low.add(per.value, per.unit)
+      current_high = current_low.add(per.value, per.unit).predecessor()
+      intervalToAdd = new dtivl.Interval(current_low, current_high, true, true)
+
     return results
 
+  truncateToPrecision: (value, unit) ->
+    # If interval boundaries are more precise than per quantity, truncate to
+    # the precision specified by the per
+    # If the per is more precise than the interval, fill out precision with
+    # minimum values so that the expansion can produce a more precise result
```

Comment:
This comment:

> \# If the per is more precise than the interval, fill out precision with
> \# minimum values so that the expansion can produce a more precise result

seems out of place here, since it's not actually doing this in this code.